portfolio/src/main/webapp/script.js

```diff
@@ -48,24 +48,44 @@ function constructImage(imageIndex){
   txtContainer.appendChild(txtElement);
 } 
 
+/*
+ * Extracts max comment from input, otherwise returns 5 
+ */
+function getMaxComment() {
+  numComments = document.getElementById('maxComments').value ;
+  return numComments || 5; 
+}
+
 /**
- * Fetches array list from the \data server and adds them to index.html
+ * Fetches comments from the \data server and adds them to index.html
  */
 function getComment() {
-  // TODO read numComments 
-  console.log("javascript, number of comments:");
-  numComments = document.getElementById('limit').value
-  // does not read numComments correctly
-  console.log(numComments);
-  fetch(`/data?numComments=5`).then(response => response.json()).then((comment) => {
+  const maxComment = getMaxComment();
+  fetch(`/data?numComments=${maxComment}`).then(response => response.json()).then((comment) => {
     // Build the list of history entries.
-    const historyEl = document.getElementById('history');
+    const commentBlock = document.getElementById('commentBlock');
+    // clear html before appending comments
+    commentBlock.innerHTML = '';
     comment.forEach((line) => {
-      historyEl.appendChild(createListElement(line.content));
+      commentBlock.appendChild(createListElement(line.content));
     });
   });
 }
 
+/*
+ * extracts numComment from URL and then displays comment
+ */
+function knit() {
+  // extract numComment from URL 
+  var url = new URL(document.URL);
+  const numComments = url.searchParams.get("numComments");
```

Comment:
nit: you should set a default value here for when a user first visits your page:
```
const numComments = url.searchParams.get("numComments") || 5;
```